src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -276,7 +276,7 @@ private byte[] classFileSetup(Consumer<? super ClassBuilder> config) {
             return ClassFile.of().build(classDesc, new Consumer<>() {
                 @Override
                 public void accept(ClassBuilder clb) {
-                    clb.withFlags(ACC_FINAL | ACC_SUPER)
```

Comment:
There's probably not much value in using ACC_FINAL here anyway.  We are only using these classes to create static methods, right?  I think ACC_INTERFACE would work here too, except for the .withSuperclass(INVOKER_SUPER_DESC) below, which maybe isn't needed.

Reply:
Note that JVMS 4.1 requires `ACC_ABSTRACT` to be also set when `ACC_INTERFACE` is set. Also note that some classes capture class data to refer to hidden classes and method handles or lambda forms, so those fields' generation need to add `ACC_PUBLIC` flag to be usable in interfaces.

Reply:
I feel like making it ACC_INTERFACE might cause some error if there are no public nonstatic methods, which is the case with this class. I don't know what @liach your comment means, but this code got more complicated than it was with the first version of this change.  When I talked to @rose00 he thought ACC_ABSTRACT would be okay for this.

Reply:
Yes, you are right that we currently don't add ACC_PUBLIC flags on methods, which will fail if we add ACC_INTERFACE. Same for the fields; these LambdaForm classes use fields to store class data that's usually stored as condy, because LambdaForm is the infrastructure that condy uses (like LambdaMetafacotry cannot use lambdas)

Reply:
Thanks for reviewing this part.